### PR TITLE
Cleanup docstring in `gammapy.maps.axes.py`

### PR DESCRIPTION
**Pull request by MRegeard**
This PR is related to issue https://github.com/gammapy/gammapy/issues/4811.

It address the part of the issue concerning `gammapy.maps.axes`.

### CODE REVIEW

File: gammapy/maps/axes.py [hunk 1]
```diff
@@ -814,15 +824,15 @@ def coord_to_pix(self, coord):
...
         Parameters
         ----------
         coord : `~numpy.ndarray`
             Array of axis coordinate values.
-        clip : bool
+        clip : bool, optional
             Choose whether to clip the index to the valid range of the
-            axis.  If false then indices for values outside the axis
+            axis. Default is False. If false then indices for values outside the axis
             range will be set -1.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:836]:
```suggestion
            axis. Default is False. If False then indices for values outside the axis
            range will be set to -1.
```

File: gammapy/maps/axes.py [hunk 2]
```diff
@@ -933,19 +943,21 @@ def copy(self, **kwargs):
         return self._init_copy(**kwargs)
 
     def round(self, coord, clip=False):
-        """Round coord to nearest axis edge.
+        """Round coord to the nearest axis edge.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:946]:
```suggestion
        """Round coordinate to the nearest axis edge.
```

File: gammapy/maps/axes.py [hunk 3]
```diff
@@ -933,19 +943,21 @@ def copy(self, **kwargs):
...
         ----------
         coord : `~astropy.units.Quantity`
-            Coordinates
-        clip : bool
-            Choose whether to clip indices to the valid range of the axis.
+            Coordinates to be rounded.
+        clip : bool, optional
+            Choose whether to clip the index to the valid range of the
+            axis. Default is False. If false then indices for values outside the axis
+            range will be set -1.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:955]:
```suggestion
            axis. Default is False. If False, then indices for values outside the axis
            range will be set to -1.
```

File: gammapy/maps/axes.py [hunk 4]
```diff
@@ -1734,7 +1746,7 @@ def drop(self, axis_name):
         Returns
         -------
         axes : `MapAxes`
-            Axes with squashed axis.
+            Axes without the axis with name `axis_name`.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:1749]: Not sure of the meaning here?

File: gammapy/maps/axes.py [hunk 5]
```diff
@@ -1771,6 +1783,10 @@ def coord_to_idx(self, coord, clip=True):
         ----------
         coord : dict of `~numpy.ndarray` or `MapCoord`
             Array of axis coordinate values.
+        clip : bool, optional
+            Choose whether to clip indices to the valid range of the axis. Default is True.
+            If False then indices for coordinates outside the axi range will be set -1.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:1788]:
```suggestion
            If False then indices for coordinates outside the axis range will be set to -1.
```

File: gammapy/maps/axes.py [hunk 6]
```diff
@@ -1810,15 +1826,15 @@ def pix_to_coord(self, pix):
         return tuple([ax.pix_to_coord(p) for ax, p in zip(self, pix)])
 
     def pix_to_idx(self, pix, clip=False):
-        """Convert pix to idx
+        """Convert pix to idx.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:1829]:
```suggestion
        """Convert pixel to index.
```

File: gammapy/maps/axes.py [hunk 7]
```diff
@@ -1810,15 +1826,15 @@ def pix_to_coord(self, pix):
...
         Parameters
         ----------
         pix : tuple of `~numpy.ndarray`
             Pixel coordinates.
-        clip : bool
+        clip : bool, optional
             Choose whether to clip indices to the valid range of the
-            axis.  If false then indices for coordinates outside
+            axis. Default is False. If false then indices for coordinates outside
             the axi range will be set -1.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:1838]:
```suggestion
            axis. Default is False. If False, then indices for coordinates outside
            the axis range will be set to -1.
```

File: gammapy/maps/axes.py [hunk 8]
```diff
@@ -1839,7 +1855,7 @@ def slice_by_idx(self, slices):
         Parameters
         ----------
         slices : dict
-            Dict of axes names and integers or `slice` object pairs. Contains one
+            Dictionary of axes names and integers or `slice` object pairs. Contains one
             element for each non-spatial dimension. For integer indexing the
             corresponding axes is dropped from the map. Axes not specified in the
             dict are kept unchanged.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:1861]:
```suggestion
            dictionary are kept unchanged.
```

File: gammapy/maps/axes.py [hunk 9]
```diff
@@ -2284,81 +2315,84 @@ def reference_time(self):
...
     @property
     def nbin(self):
-        """Return number of bins in the axis."""
+        """Return the number of bins in the axis."""
         return len(self.edges_min.flatten())
 
     @property
     def edges_min(self):
-        """Return array of bin edges max values."""
+        """Return the array of bin edges max values."""
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:2328]:
```suggestion
        """Return the array of bin edges maximum values."""
```

File: gammapy/maps/axes.py [hunk 10]
```diff
@@ -2284,81 +2315,84 @@ def reference_time(self):
...
         return self._edges_min
 
     @property
     def edges_max(self):
-        """Return array of bin edges min values."""
+        """Return the array of bin edges min values."""
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:2333]:
```suggestion
        """Return the array of bin edges minimum values."""
```

File: gammapy/maps/axes.py [hunk 11]
```diff
@@ -2284,81 +2315,84 @@ def reference_time(self):
...
     def time_format(self, val):
+        """The time format to use for the axis."""
         if val not in ["iso", "mjd"]:
             raise ValueError(f"Invalid time_format: {self.time_format}")
         self._time_format = val
 
     @property
     def as_plot_xerr(self):
-        """Plot x error"""
+        """Return x errors as if plotting."""
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:2395]: You've used this wording `as if plotting` a number of times, but it is not so clear.
My suggestion would be `for plotting`

File: gammapy/maps/axes.py [hunk 12]
```diff
@@ -3007,8 +3047,8 @@ def pix_to_coord(self, pix):
         idx = np.round(pix).astype(int)
         return self._labels[idx]
 
-    def coord_to_idx(self, coord, **kwargs):
-        """Transform labels to indices
+    def coord_to_idx(self, coord):
+        """Transform labels to indices.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:3051]: label to index or coordinate to index?

This need to be clarified throughout the docstring

File: gammapy/maps/axes.py [hunk 13]
```diff
@@ -3032,7 +3072,7 @@ def coord_to_idx(self, coord, **kwargs):
         return np.argmax(is_equal, axis=-1)
 
     def coord_to_pix(self, coord):
-        """Transform from axis labels to pixel coordinates.
+        """Transform axis labels to pixel coordinates.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:3075]: Same comment as above

File: gammapy/maps/axes.py [hunk 14]
```diff
@@ -3053,9 +3093,9 @@ def pix_to_idx(self, pix, clip=False):
         ----------
         pix : tuple of `~numpy.ndarray`
             Pixel coordinates.
-        clip : bool
+        clip : bool, optional
             Choose whether to clip indices to the valid range of the
-            axis.  If false then indices for coordinates outside
+            axis. Default is False. If False then indices for coordinates outside
             the axi range will be set -1.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:3099]:
```suggestion
            the axis range will be set to -1.
```

**Review by Astro-Kirsty — COMMENTED** [review]
Thanks @MRegeard!
I've left some comments

**Comment by Astro-Kirsty** [comment]
Another comment, is that 'oversample' and 'upsample' should be full words, not hyphenated?

**Comment by MRegeard** [comment]
> Any idea why theCI is failing because of a `clip` argument @MRegeard ?

Yes, it was because I had removed the `**kwargs` in a few method (`coord_to_pix` and `coord_to_idx`) of `TimeMapAxis` and `LabelMapAxis` because they are not used in the actual method. But they are used in `MapAxes` to deal with any kind of `Axis` inside this class. So that's why it was breaking. 
I put the `**kwargs` back !

File: gammapy/maps/axes.py (hunk 4 above)
Comment (MRegeard) [reply gammapy/maps/axes.py:1749]: It means that it returns a `MapAxes` without the axis that the user asked to `drop` by using the name of this axis to remove it. 
maybe `Axes without the axis nameed axis_name.` 
Or if you have any other wording proposition !

File: gammapy/maps/axes.py (hunk 12 above)
Comment (MRegeard) [reply gammapy/maps/axes.py:3051]: Well the thing is that `LabelMapAxis` basically is a container for string representation (using our MapAxis framework let say). But then, coordinates don't really exist... At least that's how I understand it. 
But we could just be more general and just say `axis coordinates` as here :
`    def pix_to_coord(self, pix):
        """Transform pixel to axis coordinates.`
What do you think about this ?

File: gammapy/maps/axes.py (hunk 4 above)
Comment (Astro-Kirsty) [reply gammapy/maps/axes.py:1749]: The wording is not completely understandable. Maybe: `Axes without the axis_name?`

File: gammapy/maps/axes.py (hunk 12 above)
Comment (Astro-Kirsty) [reply gammapy/maps/axes.py:3051]: I think it's more than the naming of the function is odd, given what it does. So maybe only changing it slightly, how about:
`Transform coordinate labels to indices`?

Comment (MRegeard) [reply gammapy/maps/axes.py:3051]: Seems good to me !

**Comment by Astro-Kirsty** [comment]
Thanks @MRegeard. Looks good to me, except there is a CI failing?

**Comment by MRegeard** [comment]
I started the test again and it pass so don't know why it failed in the first place.

File: gammapy/maps/axes.py [hunk 15]
```diff
@@ -2906,19 +2945,19 @@ def to_gti(self):
...
-        """Create FITS header
+        """Create FITS header.
 
         Parameters
         ----------
         format : {"ogip"}
-            Format specification
-        idx : int
-            Column index of the axis.
+            Format specification. Default is "ogip".
```
Comment (cgalelli) [inline gammapy/maps/axes.py:2953]: Isn't it "gadf" in the def?

File: gammapy/maps/axes.py [hunk 16]
```diff
@@ -3139,14 +3180,14 @@ def format_plot_xaxis(self, ax):
...
```
Comment (cgalelli) [inline gammapy/maps/axes.py:3188]: Also here

**Review by cgalelli — APPROVED** [review]
Thanks @MRegeard , I left only one comment (in two places).

File: gammapy/maps/axes.py (hunk 15 above)
Comment (MRegeard) [reply gammapy/maps/axes.py:2953]: Thanks, good catch !

**Review by cgalelli — APPROVED** [review]

File: gammapy/maps/axes.py [hunk 17]
```diff
@@ -313,33 +314,33 @@ def as_plot_labels(self):
...
     def as_plot_scale(self):
-        """Plot axis scale"""
+        """Plot axis scale."""
         mpl_scale = {"lin": "linear", "sqrt": "linear", "log": "log"}
 
         return mpl_scale[self.interp]
 
     def to_node_type(self, node_type):
-        """Return MapAxis copy changing its node type to node_type.
+        """Return a copy of the `MapAxis` instance with a node type set to node_type.
```
Comment (Astro-Kirsty) [inline gammapy/maps/axes.py:333]:
```suggestion
        """Return a copy of the `MapAxis` instance with a node type set to ``node_type``.
```

**Review by Astro-Kirsty — COMMENTED** [review]
Thanks @MRegeard. 

The final correction would be to change down-sample and up-sample to have no hyphens!

**Comment by MRegeard** [comment]
@Astro-Kirsty Done !

**Review by Astro-Kirsty — DISMISSED** [review]

File: gammapy/maps/axes.py [hunk 18]
```diff
@@ -3073,47 +3113,48 @@ def pix_to_idx(self, pix, clip=False):
...
     def center(self):
-        """Center of the label axis"""
+        """Center of the label axis."""
         return self._labels
 
     @property
     def edges(self):
-        """Edges of the label axis"""
+        """Edges of the label axis."""
+        # TODO: Why not return self._labels here?
```
Comment (registerrier) [inline gammapy/maps/axes.py:3122]:
```suggestion
```
because labels are centers

**Review by registerrier — APPROVED** [review]
Thanks @MRegeard

File: gammapy/maps/axes.py (hunk 18 above)
Comment (MRegeard) [reply gammapy/maps/axes.py:3122]: Fair enough!

Comment (MRegeard) [reply gammapy/maps/axes.py:3122]: Oh but then why are `edges_min` and `edges_max` returning `self._labels`?

Comment (registerrier) [reply gammapy/maps/axes.py:3122]: Hmm, not sure :)

**Pull request merged**